src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -14075,6 +14075,9 @@ void Unit::ProcDamageAndSpellFor(bool isVictim, Unit* target, uint32 procFlag, u
     // Player is loaded now - do not allow passive spell casts to proc
     if (GetTypeId() == TYPEID_PLAYER && ToPlayer()->GetSession()->PlayerLoading())
         return;
+    // Player is dead - do not proc on dead player
```

Comment:
If it's only for players then you should make sure that "this" is a player and not something else. Also, hi Jackpoz.